### PR TITLE
feat(cfb): add paging to question list

### PR DESCRIPTION
**Pull request by Yelinz**
image

The paging wasnt done with an infinte scroll but a button to manually load more questions on click

### CODE REVIEW

**Review by anehx — CHANGES_REQUESTED** [review]
Make it a link, this doesn't appear as actionable for users.

**Comment by Yelinz** [comment]
> Make it a link, this doesn't appear as actionable for users.

image

File: addon/templates/components/cfb-form-editor/question-list.hbs [hunk 1]
```diff
@@ -40,10 +40,7 @@
       placeholder="{{t "caluma.form-builder.global.search"}}..."
       value={{search}}
       oninput={{action
-        (queue
-          (action (mut search))
-          (action (perform data))
-        )
+        "search"
         value="target.value"
       }}
```
Comment (anehx) [inline addon/templates/components/cfb-form-editor/question-list.hbs:45]: Use the `{{on}}` helper

File: addon/templates/components/cfb-form-editor/question-list.hbs [hunk 2]
```diff
@@ -96,5 +93,8 @@
         </p>
       </li>
     {{/each}}
+    {{#if (and this.hasNextPage (eq this.mode "add"))}}
+      <li class="uk-text-center cfb-pointer"><a {{on "click" (perform this.data)}} href="#">{{t "caluma.form-builder.question.loadMore"}}</a></li>
```
Comment (anehx) [inline addon/templates/components/cfb-form-editor/question-list.hbs:97]: You'll need to prevent the default action of the event in your task

**Review by anehx — CHANGES_REQUESTED** [review]
Some nitpicks, otherwise good work!

File: addon/templates/components/cfb-form-editor/question-list.hbs (hunk 1 above)
Comment (Yelinz) [reply addon/templates/components/cfb-form-editor/question-list.hbs:45]: I would need to update the component to octane to do that

Comment (Yelinz) [reply addon/templates/components/cfb-form-editor/question-list.hbs:45]: I would need to update the component to octane to do that

File: addon/templates/components/cfb-form-editor/question-list.hbs (hunk 2 above)
Comment (Yelinz) [reply addon/templates/components/cfb-form-editor/question-list.hbs:97]: Done

File: addon/components/cfb-form-editor/question-list.js [hunk 3]
```diff
@@ -1,106 +1,112 @@
...
+  }
+  set search(event) {
+    this._search = event.target.value;
+    this._resetParameters();
+    this.questionTask.perform();
+  }
+
+  @lastValue("questionTask") questionTaskValue = [];
+  @restartableTask
+  *questionTask(caller) {
```
Comment (anehx) [inline addon/components/cfb-form-editor/question-list.js:54]: Name this `event` please

File: addon/templates/components/cfb-form-editor/question-list.hbs [hunk 4]
```diff
@@ -1,100 +1,100 @@
...
+      {{#each this.questions as |item|}}
+        <CfbFormEditor::QuestionList::Item
+          data-test-question-list-item={{item.node.slug}}
+          @mode={{this.mode}}
+          @question={{item.node}}
+          @on-edit-question={{@on-edit-question}}
+          @on-remove-question={{perform this.removeQuestion}}
+          @on-add-question={{perform this.addQuestion}}
+          @on-register={{fn this.registerChild}}
+          @on-unregister={{fn this.unregisterChild}}
```
Comment (anehx) [inline addon/templates/components/cfb-form-editor/question-list.hbs:78]: `fn` is obsolete if you're not adding any parameters

File: addon/templates/components/cfb-form-editor/question-list.hbs [hunk 5]
```diff
@@ -1,100 +1,100 @@
...
+        />
+      {{else}}
+        <li
+          data-test-question-list-empty
+          class="uk-text-small uk-text-muted uk-text-center"
+        >
+          {{uk-icon "search" ratio=4}}
```
Comment (anehx) [inline addon/templates/components/cfb-form-editor/question-list.hbs:85]:
```suggestion
          <UkIcon @icon="search" @ratio={{4}} />
```

**Review by anehx — CHANGES_REQUESTED** [review]

File: addon/templates/components/cfb-form-editor/question-list.hbs (hunk 5 above)
Comment (Yelinz) [reply addon/templates/components/cfb-form-editor/question-list.hbs]: Changed

File: addon/templates/components/cfb-form-editor/question-list.hbs (hunk 4 above)
Comment (Yelinz) [reply addon/templates/components/cfb-form-editor/question-list.hbs]: Changed

File: addon/components/cfb-form-editor/question-list.js (hunk 3 above)
Comment (Yelinz) [reply addon/components/cfb-form-editor/question-list.js]: Changed

**Review by anehx — APPROVED** [review]

**Pull request merged**